### DISCUSSION TITLE
Make it possible in the JS Libary to just get the URL

### DISCUSSION THREAD

**Discussion by MonsterUnterBett** [fief-dev/fief#175] [Ideas]
Hello all,

It would be very good to have the possibility to get the login URL in the JavaScript library returned by a function. Similar to the implementation of the Python library. Unless I have completely missed it, this function does not exist yet.

I have a real use case where it is not possible to redirect the browser directly to the login page. 

The first case is a fairly large web app where users don't have to log in from the start. By using the current implementation, it would have to be completely reloaded a second time. With my proposal, it would be possible to embed the login page (assuming the appropriate http header) in a frame and then redirect it to a small HTML page to read out the success of the login and, if applicable, the auth token. 

For the implementation I would move most of the code redirectToLogin method into a new method Namesn getLoginURL which takes the same parameters. To remain downwards copyable and to reduce the maintenance effort, I would rewrite the current redirectToLogin so that it passes its parameters to the new method and makes the redirect its return value.

**Comment by MonsterUnterBett** [comment]
Sorry i postet the Text in the wrong language.

**Comment by frankie567** [comment]
Hi @MonsterUnterBett 👋

Actually, this method exists, but it's indeed not clearly exposed in the documentation.

You can read about it in the reference: https://fief-dev.github.io/fief-js/classes/index.Fief.html#getAuthURL

I think it fulfills your need 😄

**Reply by MonsterUnterBett** [reply]
Thank you for your answer. It Looks realy promessing. I will try it out.

If you like i could write a minimal example for the documentation.

**Reply by frankie567** [reply]
Yes, I would be happy to review a PR showing a base example for JS, similar to [the one we show for Python](https://docs.fief.dev/integrate/python/).
